dlt/destinations/sql_jobs.py

```diff
@@ -749,20 +749,20 @@ def gen_scd2_sql(
             INSERT INTO {root_table_name} ({col_str}, {from_}, {to})
             SELECT {col_str}, {boundary_ts} AS {from_}, {active_record_literal} AS {to}
             FROM {staging_root_table_name} AS s
-            WHERE {hash_} NOT IN (SELECT {hash_} FROM {root_table_name});
+            WHERE {hash_} NOT IN (SELECT {hash_} FROM {root_table_name} WHERE {is_active_clause});
```

Comment:
This change enables reinsertion of a record if there is no active version of that record in the destination table.

One possible issue: users might have a UNIQUE constraint on the `_dlt_id` column in their existing `scd2` tables. This will cause a uniqueness error. Those users would have to drop those constraints manually when upgrading to a newer version of `dlt`.